### PR TITLE
AJAX Class and other fixes

### PR DESCRIPTION
**Pull request by mikejolley**
- Finishes up AJAX CRUD classes
- Fixes a bug with order item meta values
- Improves variation display in cart
- Removes attribute_ prefixes in variation class

### CODE REVIEW

File: includes/data-stores/class-wc-product-variation-data-store-cpt.php [hunk 1]
```diff
@@ -121,7 +121,7 @@ public function create( &$product ) {
 	public function update( &$product ) {
 		$post_data = array(
 			'ID'             => $product->get_id(),
-			'post_title'     => get_the_title( $product->get_parent_id() ) . ' &ndash;' . wc_get_formatted_variation( $product->get_attributes(), true ),
+			'post_title'     => get_the_title( $product->get_parent_id() ) . ' &ndash;' . wc_get_formatted_variation( $product, true, false ),
```
Comment (claudiosanches) [inline includes/data-stores/class-wc-product-variation-data-store-cpt.php]: Missing space here?

File: includes/abstracts/abstract-wc-legacy-product.php [hunk 2]
```diff
@@ -461,18 +461,6 @@ public function get_post_data() {
 	}
 
 	/**
-	 * Get the title of the post.
-	 *
-	 * @deprecated 2.7.0
-	 * @return string
-	 */
-	public function get_title() {
```
Comment (claudiosanches) [inline includes/abstracts/abstract-wc-legacy-product.php]: This should be deprecated, since products should have names and not titles.

**Review by claudiosanches — CHANGES_REQUESTED** [review]
Not a fan or products having "titles", not to mention that this forced you added a new method inside variations and introduced a new filter for it that doesn't exists in past versions...

We always used it inside the `WC_Product`:

```php
	public function get_title() {
		return apply_filters( 'woocommerce_product_title', $this->post ? $this->post->post_title : '', $this );
	}
```

So if someone is trying to change a variation title with this filter will not be possible anymore with this change now.
We are breaking backwards compatibility to keep a method that should be deprecated....

**Comment by mikejolley** [comment]
I'll fix up the hook name and add a comment. We kinda need the ability to get a parents title from a variation, so rather than introduce a new method I've jus kept the old one around. Discuss on slack with me later.

**Pull request merged**